node-red-contrib-botbuilder/msbot-connect.js

```diff
@@ -43,35 +93,45 @@ module.exports = function(RED) {
 // ------------------------------------------
 // SERVER
 // ------------------------------------------
-
-let REPLY_HANDLER = {};
-let server;
-
 async function start(node, config, RED) {
-
   if (REPLY_HANDLER[node.id]) {
     helper.removeListener("reply", REPLY_HANDLER[node.id]);
   }
   
   server = RED.httpNode;
+  
+  try {
+    let { handleReceive, reply, skillEndpoint, bot } = await initConnector(config, node, allowedCallers);
+    module.exports.botbuilder = bot;
 
-  let { handleReceive, reply } = await initConnector(config, node, config.startCmd)
+    server.get("/api/messages", (req, res, next) => {
+      res.send("Hello I'm a bot !");
+      return next();
+    });
 
-  server.get("/server-botbuilder", (req, res, next) => {
-    res.send("Hello I'm a Bot !");
-    return next();
-  });
+    // bot framework v4 messaing endpoint
+    server.post("/api/messages", (req, res) => {
+      handleReceive(req, res);
+    });
 
-  server.post("/server-botbuilder", (req, res) => {
-    handleReceive(req, res);
-  });
-  
-  node.status({ fill: "green", shape: "dot", text: "connected" });
+    // The bot defines an endpoint that forwards incoming skill activities to the root bot's skill handler (skill host endpoint)
+    if (config.botType === 'rootBot' && skillEndpoint) {
+      // expose skill host endpoint
+      server.post("/api/skills/v3/conversations/:conversationId/activities/:activityId");
```

Comment:
this server.post is useless, this route is already added by skillEndpoint.register(server, '/api/skills');

Reply:
To test first if it works without.

Reply:
Tested, it works without.